### DISCUSSION TITLE
ptrack File Not Getting Created

### DISCUSSION THREAD

**Discussion by Sasa-fras** [nrc-cnrc/EGSnrc#1119] [Questions] [answered]
I just realized that for who knows how long, my input file stopped creating new ptrack files when I run it. I have just been viewing a single ptrack file that hasn't been overwritten for who knows how many modifications to my simulated geometry.  Is there any list of issues that could just make a ptrack file not get created? Is there anywhere I could find a error saying that it wasn't created and why? Because I haven't gotten any errors saying it cant make one in the terminal output. Incase it matters, my geometry is mostly composed of ndgeometry cylinders, a cone stack, two boxes and a few union geometries. Is it some issue like you can only have one union geometry or you can't have unused geometries defined? I honestly have no clue how to troubleshoot this.

⬇️⬇️⬇️⬇️⬇️
I think it might have to do with running the simulation on multiple cores because I do see new and correct ptrack files for each core but they are not getting joined into a single ptrack file at the end like how the egslog file is.
⬆️⬆️⬆️⬆️⬆️

**Comment by rtownson** [comment, marked as answer]
It's a parallelization issue - you're not meant to have tracks being generated when you run in parallel since you can generate as large of a ptracks file as you could possibly want in a minute or less on a single core. My recommendation is just to turn off track generation when you parallelize. We should improve the messaging/documentation about this.

**Reply by Sasa-fras** [reply]
As a temporary solution I've just been running it in parallel to create an egslog file for my energy deposition data and separately running it on one core to see it with the ptracks. It is considerably faster on 8 cores then on 1. Also the 'stop scoring' isn't doing anything for some reason, i set it to something but it will seemingly do all of the tracks, like 18 thousand when i only need like 2 thousand.
